scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -949,14 +954,28 @@ static RuntimeException launderThrowable(Throwable t) {
         }
     }
 
-    void terminateJobHandling(final JobId jobId) {
+    void terminateJobHandling(final JobId jobId, final Map<String, String> jobGenericInfo) {
         try {
             listenJobLogsSupport.cleanLoggers(jobId);
             jlogger.close(jobId);
 
-            // auto remove
+            long removeDelay = Long.MAX_VALUE;
             if (SchedulingService.SCHEDULER_AUTO_REMOVED_JOB_DELAY > 0) {
-                long timeToRemove = System.currentTimeMillis() + SchedulingService.SCHEDULER_AUTO_REMOVED_JOB_DELAY;
+                removeDelay = SchedulingService.SCHEDULER_AUTO_REMOVED_JOB_DELAY;
+            }
+            if (jobGenericInfo != null && jobGenericInfo.containsKey(GENERIC_INFO_REMOVAL_DELAY)) {
+                try {
+                    removeDelay = Math.min(removeDelay,
+                                           Tools.formatDate(jobGenericInfo.get(GENERIC_INFO_REMOVAL_DELAY)));
+                } catch (Exception e) {
+                    logger.error("Error when parsing generic information " + GENERIC_INFO_REMOVAL_DELAY + " for job " +
```

Comment:
I would advice to remind in the log the value of the default removal delay to ease the log exploitation in debugging session.

Reply:
It is not necessary as I now print in INFO the exact data when the job is removed